### ISSUE TITLE
Is this feature in your road map?

### ISSUE THREAD

**Issue by bkkite** [google-ar/sceneform-android-sdk#325]
We have a server where users can load building projects. Each project is composed of some files (DOCS, PDF, GLTF, ...). These GLTF has all 3D information and some extras. What we want is to **(1)download this files** (gltf) **(2)load them into java objects** and then reproduce inside an anchor and **(3)creating a node's** tree. 

1 -> Completed, 
2-> Completed 
3-> I have no idea how to do that with ARCore. 

We have **done in IOS with ARKit** because SceneKit has a similar structure than Gltf, but the only way that found to charge a node with 3D information is node.setRenderable() with a static file, or like Uri in a server. But nothing like node.setMesh, node.setPrimitives, ... or other basic's methods to load 3D information  in a node.

Is this feature in your road map?

Thanks!.

**Comment by gstanlo** [comment]
To clarify, you'd like to render a glTF file at runtime?

**Comment by bkkite** [comment]
What we want is to build a scene with the information that comes in a gltf file. Creating as many nodes in ARCore as nodes exist in the gltf.

We use the gltf format to represent 3D information and save data associated with those nodes within the extra field of each node. To then be able to associate a listener to each node in ArCore and be able to show that information extra to the user.

As i wrote in initial post we have done with ARKit, but we don't found any way to do that with ARCore.

**Comment by dsternfeld7** [comment]
Are you reading the data of a gltf file manually? You can create Renderables for the objects within the gltf file using the class RenderableDefinition.

Right now, if you load a gltf file by converting it to a .sfb the hierarchy of the gltf file is flattened and the extra field is lost. If you aren't able to do this using RenderableDefinition, we can look into this as a feature request.

**Comment by bkkite** [comment]
> Are you reading the data of a gltf file manually?

Yes we done! but not for pleasure. You wrote the reason above "_Right now, if you load a gltf file by converting it to a .sfb the hierarchy of the gltf file is flattened and the extra field is lost._"

> You can create Renderables for the objects within the gltf file using the class RenderableDefinition.

We tried, but RenderableDefinition has simple and really basic API and we have a lot of questions about how to do that... Maybe if you make open source the pluggin's code (gltf->sbf) we could do somethig or reading some examples about how to do it, but for now is a great effort for us and we hesitate that could accomplish it in a reasonable period of time.

>we can look into this as a feature request.

Read extra fields of gltf in a ARCore node after converting sounds like Magic!!!  It could be great !!!

**Comment by bkkite** [comment]
> we can look into this as a feature request.

Will be a new feature?

**Comment by malik-at-work** [comment]
Sorry for the long response, the delay is that we can't give roadmap / feature plans via this forum.  When we add a feature the first comment on it here will be when the code is committed.

**Comment by faizan1990** [comment]
> Are you reading the data of a gltf file manually? You can create Renderables for the objects within the gltf file using the class RenderableDefinition.

@dsternfeld7 can you please give an example how?
